sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -32,102 +32,51 @@
-            result_data = []
-            result_index = []
-
-            for dataframe_item in self._data:
-                item_count = dataframe_item.count()
-                item_name = dataframe_item._name
-                result_data.append(item_count)
-                result_index.append(item_name)
-
-            return pandas.Series(data=result_data, index=result_index)
-
-        return sdc_pandas_dataframe_count_impl
-
 else:
-    def sdc_pandas_dataframe_reduce_columns(df, name, param):
+    def sdc_pandas_dataframe_reduce_columns_series(df, name, params):
         saved_columns = df.columns
         n_cols = len(saved_columns)
         data_args = tuple('data{}'.format(i) for i in range(n_cols))
-        help_param = ', {}={}):'
-        func_text = 'def _reduce_impl(df):'
-        all_params = ['df']
-        for name, value in param:
-            all_params.append('{}={}'.format(name, value))
-        func_definition = 'def _reduce_impl({}):'.format(', '.join(all_params))
+        space = []
+        if len(params) > 0:
+            space.append(', ')
+        func_definition = 'def _reduce_impl(df{}{}):'.format("".join(space), ", ".join(
+            str(key) + '=' + str(value) for key, value in params))
         func_lines = [func_definition]
         for i, d in enumerate(data_args):
-            line = '  {} = hpat.hiframes.api.init_series(hpat.hiframes.pd_dataframe_ext.get_dataframe_data(df, {}))'
+            line = '  {} = sdc.hiframes.api.init_series(sdc.hiframes.pd_dataframe_ext.get_dataframe_data(df, {}))'
             func_lines.append(line.format(d + '_S', i))
-            func_lines.append('  {} = {}.{}()'.format(d + '_O', d + '_S', name))
-        func_lines.append('  data = np.array(({},))'.format(
-            ", ".join(d + '_O' for d in data_args)))
-        func_lines.append('  index = hpat.str_arr_ext.StringArray(({},))'.format(
-            ', '.join('"{}"'.format(c) for c in saved_columns)))
-        func_lines.append('  return hpat.hiframes.api.init_series(data, index)')
+            func_lines.append('  {} = {}.{}({})'.format(d + '_O', d + '_S', name, ", ".join(
+                str(key) for key, value in params)))
+        func_lines.append("  return sdc.hiframes.pd_dataframe_ext.init_dataframe({}, None, {})\n".format(
+            ", ".join(d + '_O._data' for d in data_args),
+            ", ".join("'" + c + "'" for c in saved_columns)))
+
         loc_vars = {}
         func_text = '\n'.join(func_lines)
-
-        exec(func_text, {'hpat': sdc, 'np': numpy}, loc_vars)
+        exec(func_text, {'sdc': sdc, 'np': numpy}, loc_vars)
         _reduce_impl = loc_vars['_reduce_impl']
 
         return _reduce_impl
 
-    def check_type(name, df, axis=None, skipna=None, level=None, numeric_only=None, ddof=1, min_count=0):
+    def check_type(name, df, axis=None, skipna=None, level=None, numeric_only=None, ddof=1, min_count=0, n=5):
```

Comment:
Is it a specific function for head method? May be it is not necessary to separate it in one case?

Reply:
Its common func for all cases that write in it